src/summary/history/allergy-card-level-three.component.tsx

```diff
@@ -0,0 +1,225 @@
+import React from "react";
+import { match } from "react-router";
+import SummaryCard from "../cards/summary-card.component";
+import style from "./allergy-card-level-three.css";
+import {
+  getAllergyAllergenByConceptUuid,
+  getAllergyReaction
+} from "./allergy-intolerance.resource";
+import { createErrorHandler } from "@openmrs/esm-error-handling";
+
+const DRUGALLERGENCONCEPT: string = "162552AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+const ENVIROMENTALALLEGENCONCEPT: string =
+  "162554AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+const FOODALLERGENCONCEPT: string = "162553AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA";
+
+export function AllergyCardLevelThreeAdd(props: AllergyCardLevelThreeAddProps) {
+  const [categoryReaction, setCategoryReaction] = React.useState(null);
+  const [allergen, setAllergen] = React.useState(null);
+  const [allergyReaction, setAllergyReaction] = React.useState();
+  const [patientAllergen, setPatientAllergen] = React.useState(null);
+  const [patientAllergenReaction, setPatientAllergenReaction] = React.useState(
+    []
+  );
+  const [
+    patientAllergenReactionSeverity,
+    setPatientAllergenReactionSeverity
+  ] = React.useState(null);
+  const [dateOfOnSet, setDateOfOnSet] = React.useState(null);
+  const [comment, setComment] = React.useState(null);
+
+  const handleCategoryReactionChange = event => {
+    setAllergen(null);
+    setCategoryReaction(event.target.value);
+  };
+
+  const handleSubmit = event => {
+    event.preventDefault();
+    alert("SUbmitting the form");
+  };
+
+  React.useEffect(() => {
+    if (categoryReaction) {
+      getAllergyAllergenByConceptUuid(categoryReaction).subscribe(
```

Comment:
I do not have a strong opinion on the correctness so do not change just because of this comment. Could you explain why you are using an observable here rather than simply fetching?